apps/server/src/shared/testing/factory/user-and-account.test.factory.ts

```diff
@@ -0,0 +1,73 @@
+import { Permission, Role, School, EntityId, Account, User, BaseEntity } from '@shared/domain';
+import { ObjectId } from 'bson';
+import _ from 'lodash';
+import { userPermissions, studentPermissions, teacherPermissions, adminPermissions } from '../user-role-permissions';
+import { accountFactory } from './account.factory';
+import { roleFactory } from './role.factory';
+import { schoolFactory } from './school.factory';
+import { userFactory } from './user.factory';
+
+export interface UserAndAccountParams {
+	username?: string;
+	systemId?: EntityId | ObjectId;
+	firstName?: string;
+	lastName?: string;
+	email?: string;
+	school?: School;
+}
+
+export class UserAndAccountTestFactory {
+	private static checkIdExists(entity: BaseEntity): void | Error {
+		if (!entity.id) {
+			throw new Error('Entity must be have a valid id.');
```

Comment:
```suggestion
			throw new Error('Entity does not have an id.');
```
Ob die ID valide ist, wird ja nicht gecheckt.